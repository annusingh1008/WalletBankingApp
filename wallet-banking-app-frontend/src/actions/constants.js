export const authConstants = {
    LOGIN_REQUEST: 'LOGIN_REQUEST',
    LOGIN_SUCCESS: 'LOGIN_SUCCESS',
    LOGIN_FAILURE: 'LOGIN_FAILURE',
    LOGOUT_REQUEST: 'LOGOUT_REQUEST',
    LOGOUT_SUCCESS: 'LOGOUT_SUCCESS',
    LOGOUT_FAILURE: 'LOGOUT_FAILURE'
}

export const userConstants = {
    USER_REGISTER_REQUEST: 'USER_REGISTER_REQUEST',
    USER_REGISTER_SUCCESS: 'USER_REGISTER_SUCCESS',
    USER_REGISTER_FAILURE: 'USER_REGISTER_FAILURE',
}

export const amountTransferConstants = {
    AMOUNT_TRANSFER_REQUEST: 'AMOUNT_TRANSFER_REQUEST',
    AMOUNT_TRANSFER_SUCCESS: 'AMOUNT_TRANSFER_SUCCESS',
    AMOUNT_TRANSFER_FAILURE: 'AMOUNT_TRANSFER_FAILURE'
}

export const rechargeConstants = {
    RECHARGE_REQUEST: 'RECHARGE_REQUEST',
    RECHARGE_SUCCESS: 'RECHARGE_SUCCESS',
    RECHARGE_FAILURE: 'RECHARGE_FAILURE',
}

export const getUserDetailsConstants = {
    GET_USER_DETAILS_REQUEST: 'GET_USER_DETAILS_REQUEST',
    GET_USER_DETAILS_SUCCESS: 'GET_USER_DETAILS_SUCCESS',
    GET_USER_DETAILS_FAILURE: 'GET_USER_DETAILS_FAILURE'
}

export const transactionsConstants = {
    TRANSACTIONS_REQUEST: 'TRANSACTIONS_REQUEST',
    TRANSACTIONS_SUCCESS: 'TRANSACTIONS_SUCCESS',
    TRANSACTIONS_FAILURE: 'TRANSACTIONS_FAILURE',
    TOTAL_TRANSACTIONS_SUCCESS: 'TOTAL_TRANSACTIONS_SUCCESS',
}

export const cashbackConstants = {
    CASHBACK_REQUEST: 'CASHBACK_REQUEST',
    CASHBACK_SUCCESS: 'CASHBACK_SUCCESS',
    CASHBACK_FAILURE: 'CASHBACK_FAILURE',
    TOTAL_CASHBACKS_SUCCESS: 'TOTAL_CASHBACKS_SUCCESS'
}

export const getBalanceConstants = {
    GET_BALANCE_SUCCESS: 'GET_BALANCE_SUCCESS'
}