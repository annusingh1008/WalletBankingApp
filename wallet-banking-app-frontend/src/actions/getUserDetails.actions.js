// import { getUserDetailsConstants } from "./constants";
// import axios from "axios";

// export const getUserDetails = (email) => {
//   return async (dispatch) => {
//     const res = await axios.get(
//       `http://localhost:8085/getUserDetails/${email}`
//     );

//     if (res.status === 200) {
//       dispatch({
//         type: getUserDetailsConstants.GET_USER_DETAILS_SUCCESS,
//         payload: res.data,
//       });
//     }
//   };
// };
