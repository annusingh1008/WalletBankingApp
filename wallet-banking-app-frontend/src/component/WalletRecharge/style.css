.container{
    margin-left: 320px;
    width: 800px;
}

.heading{
    text-align: center;
    margin-left: 180px;
    width: 350px;
    margin-top: 80px;
    font-weight: bold;
    font-size: 20px;
    background-color: rgba(4, 1, 26, 0.933);
    color: white;
}

.form{
    margin-left: 200px;
}

.form label{
    margin-top: 20px;
}

.form input{
    margin-top: 10px;
    width: 310px;
}

.form button{
    margin-top: 25px;
}