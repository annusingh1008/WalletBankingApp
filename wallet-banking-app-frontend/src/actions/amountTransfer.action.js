import axios from "axios";
import { amountTransferConstants } from "./constants";

export const amountTransfer = (amountDetails) => {
  return async (dispatch) => {
    const res = await axios.post("http://localhost:8085/transfer/amount", {
      ...amountDetails,
    });

    if (res.status === 200) {
      if (res.data === "User does not exist") {
        alert("User does not exist! \nPlease check your email.");
      } else if (res.data === "Amount Transferred Successfully...!!") {
        alert("Transaction Successful");

        dispatch({
          type: amountTransferConstants.AMOUNT_TRANSFER_SUCCESS,
          payload: {
            amount:
              parseInt(amountDetails.currentAmount) -
              parseInt(amountDetails.amount),
            email: amountDetails.email,
          },
        });
      }
    }
  };
};
