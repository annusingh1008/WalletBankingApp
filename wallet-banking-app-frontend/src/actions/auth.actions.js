import axios from "axios";
import { authConstants } from "./constants";

export const login = (user) => {
  return async (dispatch) => {
    const res = await axios.post("http://localhost:8085/signin", {
      ...user,
    });

    if (res.status === 200) {
      if (res.data === "Signed in Successfully...!!") {
        localStorage.setItem("user", JSON.stringify(user));
        localStorage.setItem("email", user.email);

        dispatch({
          type: authConstants.LOGIN_SUCCESS,
          payload: user,
        });
      } else if (res.data === "Account does not exists with this email...!!") {
        alert("Account does not exists!! \nPlease check your email..");
      } else if (res.data === "Incorrect Password...!!") {
        alert("Please check your Password!!");
      }
    }
  };
};

export const getAuthDetails = (state) => state.auth;

export const signout = () => {
  return async (dispatch) => {
    dispatch({ type: authConstants.LOGOUT_REQUEST });
    localStorage.clear();
    dispatch({ type: authConstants.LOGOUT_SUCCESS });
  };
};
