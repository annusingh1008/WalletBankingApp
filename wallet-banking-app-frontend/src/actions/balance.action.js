import axios from "axios";
import { getBalanceConstants } from "./constants";

export const getUserBalance = (email) => {
  return async (dispatch) => {
    const res = await axios.get(`http://localhost:8085/getBalance/${email}`);

    dispatch({
      type: getBalanceConstants.GET_BALANCE_SUCCESS,
      payload: res.data,
    });
  };
};
