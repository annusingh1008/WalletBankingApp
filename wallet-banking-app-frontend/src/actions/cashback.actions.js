import axios from "axios";
import { cashbackConstants } from "./constants";

export const getAllCashBacks = (currentPage, email) => {
  return async (dispatch) => {
    const userEmail = localStorage.getItem("email");
    const res = await axios.get(
      `http://localhost:8085/getAllCashbacks/${userEmail}?pageNumber=${currentPage}&pageSize=${10}`
    );

    dispatch({
      type: cashbackConstants.CASHBACK_SUCCESS,
      payload: res.data,
    });
  };
};

export const getCashbacks = (state) => state.cashbacks;

export const getTotalCashbacks = (email) => {
  return async (dispatch) => {
    const res = await axios.get(
      `http://localhost:8085/getTotalCashbacks/${email}`
    );

    dispatch({
      type: cashbackConstants.TOTAL_CASHBACKS_SUCCESS,
      payload: res.data,
    });
  };
};
