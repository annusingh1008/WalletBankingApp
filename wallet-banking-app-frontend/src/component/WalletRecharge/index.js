import React, { useState } from "react";
import { Container, Row, Col, Form, Button } from "react-bootstrap";
import { useDispatch, useSelector } from "react-redux";
import { recharge } from "../../actions/recharge.actions";
import { getBalance } from "../../reducers/balance.reducer";
import Layout from "../Layout";
import "./style.css";

const WalletRecharge = () => {
  const [amount, setAmount] = useState();
  const [amountErr, setAmountErr] = useState("");

  const dispatch = useDispatch();
  // const balance = useSelector((state) => state.balance);
  const balance = useSelector(getBalance);

  const userEmail = localStorage.getItem("email");

  const walletRecharge = (e) => {
    e.preventDefault();
    if (amount <= 0) {
      setAmountErr("Enter a valid amount");
    }

    const rechargeDetails = {
      email: userEmail,
      amount: amount,
      currentAmount: balance.amount,
    };

    dispatch(recharge(rechargeDetails));
  };

  return (
    <Layout sidebar>
      <Container className="container">
        <Row md={12}>
          <Col>
            <div className="heading">Wallet Recharge</div>
            <Form
              data-testid="form"
              className="form"
              onSubmit={(e) => walletRecharge(e)}
            >
              <label>Amount</label>
              <br />
              <input
                data-testid="amountInputBox"
                type="number"
                placeholder="1000"
                value={amount}
                onChange={(e) => setAmount(e.target.value)}
                required
              ></input>
              <br />
              {amountErr && (
                <p data-testid="error-msg" className="text-danger">
                  Please enter a valid amount
                </p>
              )}
              <Button data-testid="button" type="submit">
                Recharge
              </Button>
            </Form>
          </Col>
        </Row>
      </Container>
    </Layout>
  );
};

export default WalletRecharge;
