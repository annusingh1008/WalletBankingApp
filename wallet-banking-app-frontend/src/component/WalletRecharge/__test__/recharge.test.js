import React from "react";
import { render, screen, fireEvent, act } from "@testing-library/react";
import WalletRecharge from "../index";
import { Provider } from "react-redux";
import store from "../../../store/index";
import { BrowserRouter as Router } from "react-router-dom";
import { recharge } from "../../../actions/recharge.actions";
import { Form } from "react-bootstrap";
import { getBalance } from "../../../reducers/balance.reducer";

const mockDispatch = jest.fn();
jest.mock("react-redux", () => ({
  ...jest.requireActual("react-redux"),
  useDispatch: () => mockDispatch,
}));

jest.mock("../../../actions/recharge.actions", () => ({
  recharge: jest.fn(() => ({ type: "recharge" })),
}));

jest.mock("../../../reducers/balance.reducer", () => ({
  getBalance: jest.fn(() => ({ amount: 1000 })),
}));

jest.mock("../../Layout", () =>
  jest.fn(({ children }) => <div data-testid="layout">{children}</div>)
);

jest.mock("react-bootstrap", () => ({
  Row: jest.fn(({ children }) => <div data-testid="row">{children}</div>),
  Col: jest.fn(({ children }) => <div data-testid="col">{children}</div>),
  Container: jest.fn(({ children }) => (
    <div data-tesid="container">{children}</div>
  )),
  Button: jest.fn(({ children }) => (
    <div data-testid="transferAmountButton">{children}</div>
  )),
  Form: jest.fn(({ children }) => <div data-testid="form">{children}</div>),
}));

const localStorageData = {
  email: "annu",
  password: "1234",
};

const localStorageMock = (function () {
  return {
    getItem: function (key) {
      return localStorageData[key];
    },
    removeItem: function () {
      return null;
    },
    setItem: jest.fn(),
  };
})();

describe("WalletRecharge", () => {
  beforeEach(() => {
    jest.clearAllMocks();
  });

  Object.defineProperty(window, "localStorage", { value: localStorageMock });

  it("should render amount input element and label", () => {
    render(
      <Provider store={store}>
        <Router>
          <WalletRecharge />
        </Router>
      </Provider>
    );

    const inputElement1 = screen.getByPlaceholderText(/1000/i);
    expect(inputElement1).toBeInTheDocument();
  });

  it("should render amount label element", () => {
    render(
      <Provider store={store}>
        <Router>
          <WalletRecharge />
        </Router>
      </Provider>
    );

    const labeElement1 = screen.getByText("Amount");
    expect(labeElement1).toBeInTheDocument();
  });

  it("should render recharge button", () => {
    render(
      <Provider store={store}>
        <Router>
          <WalletRecharge />
        </Router>
      </Provider>
    );

    const buttonElement = screen.getByText("Recharge");
    expect(buttonElement).toBeInTheDocument();
  });

  it("renders properly", () => {
    const { container } = render(
      <Provider store={store}>
        <Router>
          <WalletRecharge />
        </Router>
      </Provider>
    );
    expect(container.firstChild).toMatchSnapshot();
  });

  describe("#getItem", () => {
    it("returns the user email", () => {
      const result = localStorage.getItem("email");
      expect(result).toEqual("annu");
    });
  });

  it("should be able to type recharge details", () => {
    render(
      <Provider store={store}>
        <Router>
          <WalletRecharge />
        </Router>
      </Provider>
    );

    const textAreaElement1 = screen.getByPlaceholderText(/1000/i);
    fireEvent.change(textAreaElement1, { target: { value: "1000" } });
    expect(textAreaElement1.value).toBe("1000");
  });

  describe("when user puts 0 amount", () => {
    it("Shows error message", () => {
      const { queryByTestId } = render(
        <Provider store={store}>
          <Router>
            <WalletRecharge />
          </Router>
        </Provider>
      );
      act(() => {
        const amountElement = screen.queryByTestId("amountInputBox");
        fireEvent.change(amountElement, { target: { value: 0 } });
        const { onSubmit } = Form.mock.calls[1][0];
        onSubmit({ preventDefault: jest.fn() });
      });
      expect(queryByTestId("error-msg")).toBeVisible();
    });
  });

  describe("when user puts valid amount", () => {
    it("dispatches proper action", () => {
      const { queryByTestId } = render(
        <Provider store={store}>
          <Router>
            <WalletRecharge />
          </Router>
        </Provider>
      );
      act(() => {
        const amountElement = screen.queryByTestId("amountInputBox");
        fireEvent.change(amountElement, { target: { value: 500 } });
        const { onSubmit } = Form.mock.calls[1][0];
        onSubmit({ preventDefault: jest.fn() });
      });
      expect(mockDispatch).toBeCalledWith(recharge());
      expect(recharge).toBeCalledWith({
        email: "annu",
        amount: "500",
        currentAmount: 1000,
      });
    });
  });
});
