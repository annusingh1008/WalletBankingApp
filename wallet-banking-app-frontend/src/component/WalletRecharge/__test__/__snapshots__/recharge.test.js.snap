// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`WalletRecharge renders properly 1`] = `
<div
  data-testid="layout"
>
  <div
    data-tesid="container"
  >
    <div
      data-testid="row"
    >
      <div
        data-testid="col"
      >
        <div
          class="heading"
        >
          Wallet Recharge
        </div>
        <div
          data-testid="form"
        >
          <label>
            Amount
          </label>
          <br />
          <input
            data-testid="amountInputBox"
            placeholder="1000"
            required=""
            type="number"
            value=""
          />
          <br />
          <div
            data-testid="transferAmountButton"
          >
            Recharge
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
